# Assignment - 2, Gurram Manoj Reddy, 1st question

In [8]:
import tensorflow as tf
Input1 = tf.Variable(0.)
with tf.GradientTape() as tape:
    value1=tf.sin(Input1)
value1=tape.gradient(value1,Input1)
print(value1)

Input2 = tf.Variable(0.1)
with tf.GradientTape() as tape:
    value2=tf.sin(Input2)
value2=tape.gradient(value2,Input2)
print(value2)

Input3 = tf.Variable(0.2)
with tf.GradientTape() as tape:
    value3=tf.sin(Input3)
value3=tape.gradient(value3,Input3)
print(value3)

Input4 = tf.Variable(0.3)
with tf.GradientTape() as tape:
    value4=tf.sin(Input4)
value4=tape.gradient(value4,Input4)
print(value4)

tf.Tensor(1.0, shape=(), dtype=float32)
tf.Tensor(0.9950042, shape=(), dtype=float32)
tf.Tensor(0.9800666, shape=(), dtype=float32)
tf.Tensor(0.9553365, shape=(), dtype=float32)


## The value of the Gradient of sin(x) when x=0 using gradient tape is 1.0
## The value of the Gradient of sin(x) when x=0.1 using gradient tape is 0.9950042
## The value of the Gradient of sin(x) when x=0.2 using gradient tape is 0.9800666
## The value of the Gradient of sin(x) when x=0.3 using gradient tape is 0.9553365